FirebaseCore/Tests/Unit/FIRConfigurationTest.m

```diff
@@ -28,4 +28,47 @@ - (void)testSharedInstance {
   XCTAssertNotNil(config.analyticsConfiguration);
 }
 
+- (void)testGetDefaultLevel {
+  FIRConfiguration *config = [FIRConfiguration sharedInstance];
+  FIRLoggerLevel defaultLevel = [config loggerLevel];
+  XCTAssertEqual(defaultLevel, FIRLoggerLevelNotice);
+}
+
+- (void)testSetAndGet {
+  FIRConfiguration *config = [FIRConfiguration sharedInstance];
+  for (FIRLoggerLevel level = FIRLoggerLevelMin; level <= FIRLoggerLevelMax; level++) {
+    [config setLoggerLevel:level];
+    XCTAssertEqual([config loggerLevel], level);
+  }
+}
+
+- (void)testSynchronizedAccess {
```

Comment:
I don't think this API is likely to be used in conditions that warrant the synchronization queue. I'm open to removing it and leaving the class unsynchronized if others agree.

Reply:
Opted to remove it because it would be a source of flaky tests, and was probably not worth the complexity.